stubs/openpyxl/openpyxl/styles/fills.pyi

```diff
@@ -1,79 +1,134 @@
-from _typeshed import Incomplete
+from _typeshed import Self, Unused
+from collections.abc import Generator, Sequence
+from typing_extensions import Final, Literal, TypeAlias
 
-from openpyxl.descriptors import Sequence
+from openpyxl.descriptors import Sequence as SequenceDescriptor
+from openpyxl.descriptors.base import _FloatSetter
+from openpyxl.descriptors.sequence import _Sequence
 from openpyxl.descriptors.serialisable import Serialisable
+from openpyxl.xml.functions import _Element
 
-FILL_NONE: str
-FILL_SOLID: str
-FILL_PATTERN_DARKDOWN: str
-FILL_PATTERN_DARKGRAY: str
-FILL_PATTERN_DARKGRID: str
-FILL_PATTERN_DARKHORIZONTAL: str
-FILL_PATTERN_DARKTRELLIS: str
-FILL_PATTERN_DARKUP: str
-FILL_PATTERN_DARKVERTICAL: str
-FILL_PATTERN_GRAY0625: str
-FILL_PATTERN_GRAY125: str
-FILL_PATTERN_LIGHTDOWN: str
-FILL_PATTERN_LIGHTGRAY: str
-FILL_PATTERN_LIGHTGRID: str
-FILL_PATTERN_LIGHTHORIZONTAL: str
-FILL_PATTERN_LIGHTTRELLIS: str
-FILL_PATTERN_LIGHTUP: str
-FILL_PATTERN_LIGHTVERTICAL: str
-FILL_PATTERN_MEDIUMGRAY: str
-fills: Incomplete
+from .colors import Color
+
+FILL_NONE: Final = "none"
+FILL_SOLID: Final = "solid"
+FILL_PATTERN_DARKDOWN: Final = "darkDown"
+FILL_PATTERN_DARKGRAY: Final = "darkGray"
+FILL_PATTERN_DARKGRID: Final = "darkGrid"
+FILL_PATTERN_DARKHORIZONTAL: Final = "darkHorizontal"
+FILL_PATTERN_DARKTRELLIS: Final = "darkTrellis"
+FILL_PATTERN_DARKUP: Final = "darkUp"
+FILL_PATTERN_DARKVERTICAL: Final = "darkVertical"
+FILL_PATTERN_GRAY0625: Final = "gray0625"
+FILL_PATTERN_GRAY125: Final = "gray125"
+FILL_PATTERN_LIGHTDOWN: Final = "lightDown"
+FILL_PATTERN_LIGHTGRAY: Final = "lightGray"
+FILL_PATTERN_LIGHTGRID: Final = "lightGrid"
+FILL_PATTERN_LIGHTHORIZONTAL: Final = "lightHorizontal"
+FILL_PATTERN_LIGHTTRELLIS: Final = "lightTrellis"
+FILL_PATTERN_LIGHTUP: Final = "lightUp"
+FILL_PATTERN_LIGHTVERTICAL: Final = "lightVertical"
+FILL_PATTERN_MEDIUMGRAY: Final = "mediumGray"
+
+fills: Final[tuple[_FillsType, ...]]
+_FillsType: TypeAlias = Literal[
+    "solid",
+    "darkDown",
+    "darkGray",
+    "darkGrid",
+    "darkHorizontal",
+    "darkTrellis",
+    "darkUp",
+    "darkVertical",
+    "gray0625",
+    "gray125",
+    "lightDown",
+    "lightGray",
+    "lightGrid",
+    "lightHorizontal",
+    "lightTrellis",
+    "lightUp",
+    "lightVertical",
+    "mediumGray",
+]
 
 class Fill(Serialisable):
     tagname: str
     @classmethod
-    def from_tree(cls, el): ...
+    def from_tree(cls: Self, el: _Element) -> Self: ...
```

Comment:
Hmm, that doesn't look right. If you're using `_typeshed.Self`, the `cls` parameter should be annotated with `type[Self]`. If you're using `typing_extensions.Self`, it doesn't need to be annotated at all.

Reply:
I need to update usages of `Self` anyway, so I'll double check this at the same time.